connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -427,4 +435,62 @@ public void ensureValid(String name, Object value) {
         }
     }
 
+    private static class ResponseHttpHeadersValidator implements ConfigDef.Validator {
+        @Override
+        public void ensureValid(String name, Object value) {
+            String strValue = (String) value;
+            if (strValue.isEmpty()) {
+                return;
+            }
+
+            String[] configs = StringUtil.csvSplit(strValue);
+            Arrays.stream(configs).forEach(config -> validateHttpResponseHeaderConfig(config));
+        }
+
+        private static void validateHttpResponseHeaderConfig(String config) {
+            try {
+                // validate format
+                String[] configTokens = config.trim().split("\\s+", 2);
+                if (configTokens.length != 2) {
+                    throw new ConfigException(String.format("Invalid format of header config \"%s\". "
+                            + "Expected: \"[ation] [header name]:[header value]\"", config));
+                }
+
+                // validate action
+                String method = configTokens[0].trim();
+                validateHeaderConfigAction(method);
+
+                // validate header name and header value pair
+                String header = configTokens[1];
+                String[] headerTokens = header.trim().split(":");
+                if (headerTokens.length != 2) {
+                    throw new ConfigException(
+                            String.format("Invalid format of header name and header value pair \"%s\". "
+                                    + "Expected: \"[header name]:[header value]\"", header));
+                }
+
+                // validate header name
+                String headerName = headerTokens[0].trim();
+                if (headerName.isEmpty() || headerName.contains(" ")) {
+                    throw new ConfigException(String.format("Invalid header name \"%s\". "
+                            + "The \"[header name]\" cannot contain whitespace", headerName));
+                }
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new ConfigException(String.format("Invalid header config \"%s\".", config), e);
+            }
+        }
+
+        private static void validateHeaderConfigAction(String action) {
+            /**
+             * The following actions are defined following link.
+             * {@link https://www.eclipse.org/jetty/documentation/current/header-filter.html}
+             **/
+            if (!Arrays.asList("set", "add", "setDate", "addDate")
+                    .stream()
+                    .anyMatch(action::equalsIgnoreCase)) {
+                throw new ConfigException(String.format("Invalid header config action: \"%s\". "
+                        + "The action need be one of [\"set\", \"add\", \"setDate\", \"addDate\"]", action));
```

Comment:
How about defining a static immutable list as a constant:
```
    private static final Collection<String> HEADER_ACTIONS = Collections.unmodifiableList(
            Arrays.asList("set", "add", "setDate", "addDate")
    );
```
so that these lines can become:
```suggestion
            if (!HEADER_ACTIONS.stream().anyMatch(action::equalsIgnoreCase)) {
                throw new ConfigException(String.format("Invalid header config action: '%s'. "
                        + "Expected one of %s", action, HEADER_ACTIONS));
```
This eliminates the duplication of the literal values (which is prone to future errors) and makes the code more readable.